synapse/storage/databases/main/receipts.py

```diff
@@ -149,10 +170,20 @@ async def get_last_receipt_event_id_for_user(
             allow_none=True,
         )
 
-    @cached(num_args=2)
+    @cached()
     async def get_receipts_for_user(
         self, user_id: str, receipt_type: str
     ) -> Dict[str, str]:
+        """
+        Fetch the event IDs for the latest receipt in all rooms with the given receipt type.
+
+        Args:
+            user_id: The user to fetch receipts for.
+            receipt_type: The receipt types to fetch.
+
+        Returns:
+            A map of room ID to the event ID of the latest receipt for that room.
```

Comment:
+159 suggests that these receipts may not exist. How is that case handled by this function?

Reply:
I'm unsure how to clarify this, the user needs to be in the room and have sent a receipt so it is returning all known receipts?

Reply:
Perhaps add something like: "If the user has joined a room and has yet to sent a receipt to it, that room will not appear in the returned dictionary." ?

Reply:
Does 0501bbc002d8d3391c3f335aa15286f4d59a6d5a help?